.github/workflows/codechecker.yaml

```diff
@@ -0,0 +1,62 @@
+name: Run Static Analysis
+
+on:
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  RunCodechecker:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.9'
+
+      - name: Install Dependencies
+        run: |
+          curl -O https://efa-installer.amazonaws.com/aws-efa-installer-1.30.0.tar.gz
+          tar -xf aws-efa-installer-1.30.0.tar.gz && cd aws-efa-installer
+          sudo ./efa_installer.sh -y
+          # https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/efa-start-nccl-base.html#nccl-start-base-drivers
+          sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
```

Comment:
Given you add multiple repos, you can use `env:` in the workflow definition to take a list of all the ones you need in a env and just add them in one shot here? `suda add-apt-repository -y ${{ env.REPOS }}`. Same for the dependency packages you install, can maintain a list and install them in one shot.

Reply:
I hit some issues with coercing those into proper bash arrays. I think it's better to just create a json file or similar under `.github/` and then find a way to query those from within the job.